go/vt/vtexplain/vtexplain_vttablet.go

```diff
@@ -75,31 +77,147 @@ func newFakeTablet() *fakeTablet {
 		dbconfigs.AppConfig, // These tests only use the app pool.
 	)
 
-	target := querypb.Target{TabletType: topodatapb.TabletType_MASTER}
+	target := querypb.Target{
+		Keyspace:   t.Keyspace,
+		Shard:      t.Shard,
+		TabletType: topodatapb.TabletType_MASTER,
+	}
 	tsv.StartService(target, dbcfgs, mysqld)
 
 	// clear all the schema initialization queries out of the tablet
 	// to avoid clutttering the output
-	tablet.queries = nil
+	tablet.mysqlQueries = nil
 
 	return &tablet
 }
 
-// Execute hook called by SandboxConn as part of running the query.
-func (tablet *fakeTablet) Execute(ctx context.Context, target *querypb.Target, query string, bindVars map[string]*querypb.BindVariable, transactionID int64, options *querypb.ExecuteOptions) (*sqltypes.Result, error) {
-	logStats := tabletenv.NewLogStats(ctx, "FakeQueryExecutor")
+var _ queryservice.QueryService = (*fakeTablet)(nil) // compile-time interface check
 
-	plan, err := tablet.tsv.GetPlan(ctx, logStats, query)
-	if err != nil {
-		return nil, err
-	}
+// Begin is part of the QueryService interface.
+func (tablet *fakeTablet) Begin(ctx context.Context, target *querypb.Target, options *querypb.ExecuteOptions) (int64, error) {
```

Comment:
You should be able to remove all these boiler-plates by anonymous embedding of tabletserver:
```
type fakeTablet struct {
  *tabletserver.TabletServer
  ...
}
```

Then just define the functions that need changing. You can call into the embedded tabletserver using: `tablet.(TabletServer).Execute(...)`

Ref: https://golang.org/ref/spec#Struct_types

Reply:
I thought about doing that but I'd rather keep the current approach that will explicitly panic if an unimplemented method gets called.

Reply:
You could embed the `ErrorQueryService` instead: https://github.com/youtube/vitess/blob/master/go/vt/vttablet/queryservice/fakes/error_query_service.go#L29

This will it will return an error for each interface method unless you override it here.

Reply:
Thanks for the suggestion -- I actually went a slightly different route with the same effect. I changed the tablet to embed an anonymous `QueryService` interface that is never initialized. This results in a runtime fault if any of the